game/hud/src/components/Progression/ProgressionView.tsx

```diff
@@ -80,53 +132,256 @@ class ProgressionView extends React.Component<Props, State> {
       return (
         <LoadingContainer>
-          <CloseButton onClick={this.props.onCloseClick} />
-          <div>{graphql.lastError}</div>
+          <ProgressionTitle><h6>Progression</h6></ProgressionTitle>
+          <InnerContainer>
+            <ProgressionCorner />
+            <CloseButton onClick={this.props.onCloseClick} />
+            <ProgressionContent>
+              <ProgressionLoading>
+                <div>{graphql.lastError}</div>
+              </ProgressionLoading>
+            </ProgressionContent>
+            <ProgressionFooter />
+          </InnerContainer>
         </LoadingContainer>
       );
     }
 
     if (graphql.loading || !graphql.data || !graphql.data.myprogression) {
       return (
         <LoadingContainer>
-          <CloseButton onClick={this.props.onCloseClick} />
-          <div>Loading...</div>
-          <Spinner />
+          <ProgressionTitle><h6>Progression</h6></ProgressionTitle>
+          <InnerContainer>
+            <ProgressionCorner />
+            <CloseButton onClick={this.props.onCloseClick} />
+            <ProgressionContent>
+              <ProgressionLoading>
+                <div>Loading...</div>
+                <Spinner />
+              </ProgressionLoading>
+            </ProgressionContent>
+            <ProgressionFooter />
+          </InnerContainer>
         </LoadingContainer>
       );
     }
 
     if (this.props.collected || isEmpty(graphql.data.myprogression.unCollectedDayLogs)) {
       return (
         <LoadingContainer>
-          <CloseButton onClick={this.props.onCloseClick} />
-          <div>All progression packages have been collected</div>
+          <ProgressionTitle><h6>Progression</h6></ProgressionTitle>
+          <InnerContainer>
+            <ProgressionCorner />
+            <CloseButton onClick={this.props.onCloseClick} />
+            <ProgressionContent>
+              <ProgressionLoading>
+                <div>All progression packages have been collected</div>
+              </ProgressionLoading>
+            </ProgressionContent>
+            <ProgressionFooter />
+          </InnerContainer>
         </LoadingContainer>
       );
     }
 
     return (
       <Container>
-        <CloseButton onClick={this.props.onCloseClick} />
-        <ObjectDisplayContainer>
-          <ObjectDisplay data={graphql.data.myprogression} skipFunctions />
+        <ProgressionTitle><h6>Progression</h6></ProgressionTitle>
+        <InnerContainer className='cse-ui-scroller-thumbonly'>
+          <CloseButton onClick={this.props.onCloseClick} />
+          <ProgressionCorner />
+          <ProgressionContent>
+
           {graphql.data.myprogression.unCollectedDayLogs.map((uncollectedDay) => {
+            const { secondsActive, distanceMoved, skillPartsUsed, damage, plots, crafting, scenarios } = uncollectedDay;
+
+            const damageDetails: JSX.Element[] = [];
+            Object.keys(damage).map((damageKey) => {
```

Comment:
These mappings and then pushing JSX elements to arrays are a tad awkward because you're using map to iterate over an array and not returning anything. There are two ways we can make this more conventional with our project.

1.) This solution is my fav and probably the way to go would be to instead of pushing things to a new array, just do
`const damageDetails = Object.keys(damage[damageKey]).map((damageType) => {
  return <li ....
})` and then you'll have an array already created with your elements.

2.) This one uses the same concept you're trying to implement here. Instead of using 'map' to iterate over these arrays, just use forEach.
`Object.keys(damage[damageKey]).forEach((damageType) => {
  damageDetails.push( ... )
})`

Reply:
Whichever you choose, just be consistent throughout the component.